### DISCUSSION TITLE
SP32 only offers a microphone profile as a Bluetooth device

### DISCUSSION THREAD

**Discussion by eric-r-moore** [pschatzmann/ESP32-A2DP#627] [Q&A] [answered]
Sorry, I'm probably new to ESP and am currently looking for a demo that can use the ESP32 as a microphone.

I was thinking that the HFP profile can transmit the sound of the phone and provide its own microphone to send to the phone. I want the phone to use its own speakers, and the ESP32 only offers microphones, is that possible?

Going one step further, I wanted the microphone's audio quality to be up to A2DP's stereo standard.

Any suggestions? Thank you very much

**Comment by pschatzmann** [comment, marked as answer]
Please check my answers to [similar questions](https://github.com/search?q=repo%3Apschatzmann%2FESP32-A2DP+HFP&type=discussions)
